src/stream/stream/zip.rs

```diff
@@ -0,0 +1,54 @@
+use std::fmt;
+use std::pin::Pin;
+
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+/// An iterator that iterates two other iterators simultaneously.
+pub struct Zip<A: Stream, B> {
+    item_slot: Option<A::Item>,
+    first: A,
+    second: B,
+}
+
+impl<A: fmt::Debug + Stream, B: fmt::Debug> fmt::Debug for Zip<A, B> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("Zip")
+            .field("first", &self.first)
+            .field("second", &self.second)
+            .finish()
+    }
+}
+
+impl<A: Unpin + Stream, B: Unpin> Unpin for Zip<A, B> {}
```

Comment:
```suggestion
impl<A: Unpin, B: Unpin> Unpin for Zip<A, B> {}
```

Reply:
It doesn't work because of `A: Stream` bound on `struct Zip`.